### ISSUE TITLE
How to specify the resolutions of the raster layer to load wmts

### ISSUE THREAD

**Issue by mwhei** [mapbox/mapbox-gl-js#12473]
Hello expert，
   I found  mapbox gl  can not load wmts with not regular scales. How to specify the resolutions of the raster layer to load wmts.? Thanks a lot.

**Comment by enersis-pst** [comment]
i don't know if this is what you are looking for?
https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/#raster-tileSize

like you told before, mapbox doesnt support wmts.

**Comment by avpeery** [comment]
Hi @mwhei there is a related issue that has been closed #6089 since wmts isn't supported. Thanks for using mapbox!

**Comment by boeckMt** [comment]
I think this could work if the server provides a `Web Mercator tileset`, or do you mean that with `regular scales`? @mwhei

```ts
const matrixSet = 'EPSG:3857';
const matrix = 'EPSG:3857:{z}';

const url = `${wmtsUrl}
?layer=${layerName}
&style=${layerStyle}
&tilematrixset=${matrixSet}
&service=WTMS
&version=1.0.0
&request=GetTile
&TileMatrix=${matrix}
&TileCol={x}
&TileRow={y}
&format=image/png

const source: SourceSpecification = {
        type: 'raster',
        tiles: [url],
};

```

Here are a link to WMTS of OpenLayers https://github.com/openlayers/openlayers/blob/v7.2.2/src/ol/source/WMTS.js#L269-L334
